src/main/java/org/jenkinsci/plugins/gitlab/GitlabBuilds.java

```diff
@@ -49,7 +54,12 @@ public String build(GitlabCause cause, Map<String, String> customParameters, Git
     	if (hasCommitStatus(project, cause.getLastCommitId(), api)) {
         	shouldRun = false;
         }
-    	
+
+        if (lastNote != null && lastNote.getBody().equals(triggerComment)) {
+            LOGGER.info("Trigger comment found");
+            shouldRun = true;
```

Comment:
OK, i see it, the `build started` message will be the last.
i think you should check `shouldRun` first then try to get the `lastNote`.
`GitlabMergeRequestWrapper` has implemented the method `getNotes`, can we reuse it?